File: backend/src/main/java/com/serenade/backend/domain/playlist/dto/PlaylistSummaryResponse.java
package com.serenade.backend.domain.playlist.dto;

import com.serenade.backend.domain.playlist.Playlist;

import java.time.Instant;
import java.util.UUID;

public record PlaylistSummaryResponse(
        UUID id,
        String name,
        boolean isCopy,
        UUID sourcePlaylistId,
        int version,
        int trackCount,
        double ratingAvg,
        Instant updatedAt
) {
    public static PlaylistSummaryResponse from(Playlist p, int trackCount, double ratingAvg) {
        return new PlaylistSummaryResponse(
                p.getId(), p.getName(), p.isCopy(), p.getSourcePlaylistId(),
                p.getVersion(), trackCount, ratingAvg, p.getUpdatedAt());
    }
}

File: app/app/src/main/java/com/serenade/app/feature/playlist/data/remote/dto/PlaylistSummaryResponse.kt
package com.serenade.app.feature.playlist.data.remote.dto

import kotlinx.serialization.Serializable

@Serializable
data class PlaylistSummaryResponse(
    val id: String,
    val name: String,
    val isCopy: Boolean,
    val sourcePlaylistId: String? = null,
    val version: Int,
    val trackCount: Int,
    val ratingAvg: Double,
    val updatedAt: String? = null,
)
